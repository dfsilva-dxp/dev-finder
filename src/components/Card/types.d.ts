import { IUser } from "context/userContext";

export interface ICardProps {
  user?: IUser | null;
}
