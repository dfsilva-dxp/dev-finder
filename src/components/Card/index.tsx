import { Buildings, LinkSimple, MapPinLine, TwitterLogo } from "phosphor-react";

import Avatar from "components/Avatar";

import {
  Bio,
  Content,
  CountDisplay,
  Counters,
  Footer,
  UserInfo,
  Username
} from "./styles";

export default function Card() {
  return (
    <Content>
      <Avatar url_img="https://avatars.githubusercontent.com/u/16245261?v=4" />

      <UserInfo>
        <Username>
          <div>
            <h1>Daniel Silva</h1>
            <a href="/link">dfsilva-dxp</a>
          </div>
          <p>Joined 2 Fev 2020</p>
        </Username>

        <Bio role="dialog" aria-labelledby="user-bio">
          <p id="user-bio">Lorem ipsum dolor sit amet, consectetur adipis</p>
        </Bio>

        <Counters>
          <CountDisplay>
            <strong>Repos</strong>
            <span>9</span>
          </CountDisplay>
          <CountDisplay>
            <strong>Followers</strong>
            <span>59</span>
          </CountDisplay>
          <CountDisplay>
            <strong>Following</strong>
            <span>43</span>
          </CountDisplay>
        </Counters>

        <Footer>
          <div>
            <span>
              <MapPinLine size={18} /> Suzano - SP
            </span>
            <span>
              <LinkSimple size={18} /> www.dfsilva.dev
            </span>
          </div>
          <div>
            <span>
              <TwitterLogo size={18} /> dfsilva.dxp
            </span>
            <span>
              <Buildings size={18} /> GWCloud
            </span>
          </div>
        </Footer>
      </UserInfo>
    </Content>
  );
}
