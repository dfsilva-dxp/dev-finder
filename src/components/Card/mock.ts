export default {
  login: "dfsilva-dxp",
  id: 16245261,
  node_id: "MDQ6VXNlcjE2MjQ1MjYx",
  avatar_url: "https://avatars.githubusercontent.com/u/16245261?v=4",
  gravatar_id: "",
  url: "https://api.github.com/users/dfsilva-dxp",
  html_url: "https://github.com/dfsilva-dxp",
  followers_url: "https://api.github.com/users/dfsilva-dxp/followers",
  following_url:
    "https://api.github.com/users/dfsilva-dxp/following{/other_user}",
  gists_url: "https://api.github.com/users/dfsilva-dxp/gists{/gist_id}",
  starred_url:
    "https://api.github.com/users/dfsilva-dxp/starred{/owner}{/repo}",
  subscriptions_url: "https://api.github.com/users/dfsilva-dxp/subscriptions",
  organizations_url: "https://api.github.com/users/dfsilva-dxp/orgs",
  repos_url: "https://api.github.com/users/dfsilva-dxp/repos",
  events_url: "https://api.github.com/users/dfsilva-dxp/events{/privacy}",
  received_events_url:
    "https://api.github.com/users/dfsilva-dxp/received_events",
  type: "User",
  site_admin: false,
  name: "Daniel Silva",
  company: "Grapeway Cloud Evolution",
  blog: "https://www.dfsilva.dev/",
  location: "Suzano - SP",
  email: null,
  hireable: true,
  bio: "Acredito que lidar com pessoas em tempos de informações rápidas, leve e inteligente me faz entender que o sucesso e as respostas estão mesmo nas perguntas.",
  twitter_username: "dfsilvadxp",
  public_repos: 122,
  public_gists: 0,
  followers: 39,
  following: 66,
  created_at: "2015-12-10T18:58:00Z",
  updated_at: "2022-10-27T11:34:56Z"
};
