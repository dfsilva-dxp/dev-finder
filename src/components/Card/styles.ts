import styled, { css } from "styled-components";

export const Content = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.colors.panel};
    border-radius: 0.625rem;
    overflow: hidden;

    display: flex;
    flex-direction: column;
  `}
`;

export const Cover = styled.header`
  ${({ theme }) => css`
    width: 100%;
    background-color: ${theme.colors.cover};
    padding: ${theme.spacings.large} ${theme.spacings.xsmall};
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
  `}
`;

export const CloseButton = styled.button`
  ${({ theme }) => css`
    background-color: transparent;
    margin-top: -1.25rem;
    border: 0;
    border-radius: ${theme.border.radius};
    color: ${theme.colors.light};
    line-height: 0;
    align-self: flex-start;

    &:hover,
    &:focus {
      outline: 2px solid ${theme.colors.primary};
      outline-offset: ${theme.border.outlineOffiset};
    }
  `}
`;

export const UserInfo = styled.header`
  ${({ theme }) => css`
    width: 100%;
    margin-top: 0.625rem;
    color: ${theme.colors.light};

    display: flex;
    align-items: center;
    flex-direction: column;

    h1 {
      font-size: ${theme.font.sizes.large};
      font-weight: ${theme.font.weight.medium};
    }

    a,
    small {
      font-weight: ${theme.font.weight.bold};
      width: fit-content;
    }

    a {
      margin-top: ${theme.spacings.xxsmall};
      margin-bottom: ${theme.spacings.xxsmall};
      border-radius: ${theme.border.radius};

      &:hover,
      &:focus {
        outline: 2px solid ${theme.colors.primary};
        outline-offset: ${theme.border.outlineOffiset};
      }
    }
  `}
`;

export const Counters = styled.div`
  ${({ theme }) => css`
    width: 100%;
    margin-top: ${theme.spacings.small};
    padding: 0 ${theme.spacings.xsmall};

    display: flex;
    align-items: center;
    justify-content: center;
    gap: ${theme.spacings.xxsmall};
  `}
`;

export const CountDisplay = styled.span`
  &,
  & a {
    ${({ theme }) => css`
      width: 7rem;
      color: ${theme.colors.light};
      padding: 0 ${theme.spacings.xxsmall};

      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;

      & + & {
        border-left: 1px solid ${theme.colors.light};
      }

      svg {
        margin-bottom: ${theme.spacings.xxsmall};
      }
      span {
        font-size: ${theme.font.sizes.xsmall};
        font-weight: ${theme.font.weight.bold};
      }
    `}
  }
`;

export const Body = styled.section`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.xsmall};
  `}
`;

export const Bio = styled.div`
  ${({ theme }) => css`
    h4 {
      font-size: ${theme.font.sizes.small};
      color: ${theme.colors.grayDark};
    }

    p {
      font-size: ${theme.font.sizes.xsmall};
      margin-top: 0.625rem;
    }
  `}
`;

export const Footer = styled.footer`
  ${({ theme }) => css`
    width: 100%;
    margin-top: 0.625rem;

    display: grid;
    grid-template-columns: repeat(2, 1fr);

    div {
      display: flex;
      flex-direction: column;
      gap: ${theme.spacings.xxsmall};

      font-size: ${theme.font.sizes.xsmall};

      span {
        display: flex;
        align-items: center;
        justify-content: flex-start;
        gap: ${theme.spacings.xsmall};
      }

      span svg {
        color: ${theme.colors.primary};
      }
    }
  `}
`;
