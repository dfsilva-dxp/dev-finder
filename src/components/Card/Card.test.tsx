import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/test/helpers";

import Card from ".";

import userMock from "./mock";

describe("<Card/>", () => {
  it("should render currectly", () => {
    const { container } = renderWithTheme(<Card user={userMock} />);

    expect(
      screen.getByRole("img", { name: /User Avatar/i })
    ).toBeInTheDocument();

    expect(
      screen.getByRole("heading", { name: userMock.name })
    ).toBeInTheDocument();

    expect(screen.getByRole("link", { name: userMock.login })).toHaveAttribute(
      "href",
      userMock.html_url
    );

    expect(screen.getByText(userMock.bio)).toBeInTheDocument();

    expect(screen.getByText(/repos/i)).toBeInTheDocument();
    expect(screen.getByText(userMock.public_repos)).toBeInTheDocument();

    expect(screen.getByText(/followers/i)).toBeInTheDocument();
    expect(screen.getByText(userMock.followers)).toBeInTheDocument();

    expect(screen.getByText(/following/i)).toBeInTheDocument();
    expect(screen.getByText(userMock.following)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
