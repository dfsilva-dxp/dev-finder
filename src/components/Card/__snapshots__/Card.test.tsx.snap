// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should render currectly 1`] = `
.c1 {
  border: 2px solid #E0E0E0;
  width: 120px;
  height: 120px;
  border-radius: 50%;
  overflow: hidden;
  padding: 4px;
}

.c1 img {
  width: 100%;
  border-radius: 50%;
}

.c0 {
  background-color: #1F2A48;
  border-radius: 0.25rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 1.5rem;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 1rem;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 h1 {
  font-size: 1.25rem;
}

.c3 a,
.c3 p {
  font-weight: 700;
  font-size: 0.75rem;
}

.c3 a {
  display: block;
  margin-top: 0.5rem;
  color: #0178FF;
}

.c3 p {
  color: #E0E0E0;
}

.c4 {
  width: 100%;
  padding: 1rem 0;
  color: #E0E0E0;
}

.c5 {
  width: 100%;
  background-color: #141C2F;
  border-radius: 0.25rem;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  gap: 0.5rem;
}

.c6 {
  padding: 0 0.5rem;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.5rem;
}

.c6 strong {
  font-size: 0.75rem;
}

.c6 span {
  font-size: 1.75rem;
  font-weight: 700;
}

.c7 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c7 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  font-size: 0.75rem;
}

.c7 div span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 1rem;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="User Avatar"
        src="https://avatars.githubusercontent.com/u/16245261?v=4"
        title="User Avatar"
      />
    </div>
    <header
      class="c2"
    >
      <div
        class="c3"
      >
        <div>
          <h1>
            Daniel Silva
          </h1>
          <a
            href="/link"
          >
            dfsilva-dxp
          </a>
        </div>
        <p>
          Joined 2 Fev 2020
        </p>
      </div>
      <div
        aria-labelledby="user-bio"
        class="c4"
        role="dialog"
      >
        <p
          id="user-bio"
        >
          Lorem ipsum dolor sit amet, consectetur adipis
        </p>
      </div>
      <div
        class="c5"
      >
        <span
          class="c6"
        >
          <strong>
            Repos
          </strong>
          <span>
            9
          </span>
        </span>
        <span
          class="c6"
        >
          <strong>
            Followers
          </strong>
          <span>
            59
          </span>
        </span>
        <span
          class="c6"
        >
          <strong>
            Following
          </strong>
          <span>
            43
          </span>
        </span>
      </div>
      <footer
        class="c7"
      >
        <div>
          <span>
            <svg
              fill="currentColor"
              height="18"
              viewBox="0 0 256 256"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <line
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="56"
                x2="200"
                y1="232"
                y2="232"
              />
              <circle
                cx="128"
                cy="104"
                fill="none"
                r="32"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M208,104c0,72-80,128-80,128S48,176,48,104a80,80,0,0,1,160,0Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
             Suzano - SP
          </span>
          <span>
            <svg
              fill="currentColor"
              height="18"
              viewBox="0 0 256 256"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="94.1"
                x2="161.9"
                y1="161.9"
                y2="94"
              />
              <path
                d="M145,178.9l-28.3,28.3a48,48,0,0,1-67.9-67.9L77.1,111"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M178.9,145l28.3-28.3a48,48,0,0,0-67.9-67.9L111,77.1"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
             www.dfsilva.dev
          </span>
        </div>
        <div>
          <span>
            <svg
              fill="currentColor"
              height="18"
              viewBox="0 0 256 256"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M128,88c0-22,18.5-40.3,40.5-40a40,40,0,0,1,36.2,24H240l-32.3,32.3A127.9,127.9,0,0,1,80,224c-32,0-40-12-40-12s32-12,48-36c0,0-64-32-48-120,0,0,40,40,88,48Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
             dfsilva.dxp
          </span>
          <span>
            <svg
              fill="currentColor"
              height="18"
              viewBox="0 0 256 256"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="16"
                x2="240"
                y1="216"
                y2="216"
              />
              <path
                d="M144,216V40a8,8,0,0,0-8-8H40a8,8,0,0,0-8,8V216"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M224,216V104a8,8,0,0,0-8-8H144"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="64"
                x2="96"
                y1="72"
                y2="72"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="80"
                x2="112"
                y1="136"
                y2="136"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="64"
                x2="96"
                y1="176"
                y2="176"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="176"
                x2="192"
                y1="176"
                y2="176"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="176"
                x2="192"
                y1="136"
                y2="136"
              />
            </svg>
             GWCloud
          </span>
        </div>
      </footer>
    </header>
  </section>
</div>
`;
